src/status_im/wallet_connect_legacy/core.cljs

```diff
@@ -159,13 +172,10 @@
              (update :wallet-connect-legacy/sessions
                      conj
                      session))
-     ::json-rpc/call [{:method     "wakuext_addWalletConnectSession"
-                       :params     [{:id peer-id
-                                     :info info
-                                     :dappName dapp-name
-                                     :dappUrl dapp-url}]
-                       :on-success #(log/info "wakuext_addWalletConnectSession success call back , data =>" %)
-                       :on-error   #(log/error "wakuext_addWalletConnectSession error call back , data =>" %)}]}))
+     ::save-session {:peer-id peer-id
```

Comment:
could you elaborate in this change? i don't understand how it will work, there is `save-session` effect is registered ?

Reply:
Instead of repeating the call to the `wakuext_addWalletConnectSession` endpoint .. And because it's used when first connecting to a dApp and when changing wallet accounts .. I extracted the logic the creates the session and now I dispatch to it from two places .. When updating the account connect too a wallet and when initially connecting to a wallet.

Reply:
but you use effect here, if you want  to use your new event you have to merge it here , `fx/merge`

Reply:
Like this?
`(fx/defn session-connected
  {:events [:wallet-connect-legacy/created]}
  [{:keys [db] :as cofx} session]
  (let [connector (get db :wallet-connect-legacy/proposal-connector)
        session (assoc (types/js->clj session)
                       :wc-version constants/wallet-connect-version-1
                       :connector connector)
        peer-id (get-in session [:params 0 :peerId])
        dapp-name (get-in session [:params 0 :peerMeta :name])
        dapp-url (get-in session [:params 0 :peerMeta :url])]
    
    (fx/merge cofx
              {:db (-> db
                      (assoc :wallet-connect/session-connected session)
                      (update :wallet-connect-legacy/sessions
                              conj
                              session))
               :show-wallet-connect-success-sheet nil}
              (save-session {:peer-id peer-id
                             :dapp-name dapp-name
                             :dapp-url dapp-url
                             :connector connector}))))`

Reply:
yes

Reply:
Cool, Thanks Andre